mws/resource_customer_managed_key.go

```diff
@@ -0,0 +1,119 @@
+package mws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/databrickslabs/databricks-terraform/common"
+	"github.com/databrickslabs/databricks-terraform/internal"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// NewCustomerManagedKeysAPI creates CustomerManagedKeysAPI instance from provider meta
+func NewCustomerManagedKeysAPI(m interface{}) CustomerManagedKeysAPI {
+	return CustomerManagedKeysAPI{client: m.(*common.DatabricksClient)}
+}
+
+// CustomerManagedKeysAPI exposes the mws customerManagedKeys API
+type CustomerManagedKeysAPI struct {
+	client *common.DatabricksClient
+}
+
+// Create creates a set of MWS CustomerManagedKeys for the BYOVPC
+func (a CustomerManagedKeysAPI) Create(mwsAcctID, keyArn, keyAlias string) (k CustomerManagedKey, err error) {
+	customerManagedKeysAPIPath := fmt.Sprintf("/accounts/%s/customer-managed-keys", mwsAcctID)
+	err = a.client.Post(customerManagedKeysAPIPath, CustomerManagedKey{
+		AwsKeyInfo: &AwsKeyInfo{
+			KeyArn:   keyArn,
+			KeyAlias: keyAlias,
+		},
+	}, &k)
+	return
+}
+
+// Read returns the customer managed key object along with metadata
+func (a CustomerManagedKeysAPI) Read(
+	mwsAcctID, customerManagedKeyID string) (k CustomerManagedKey, err error) {
+	err = a.client.Get(fmt.Sprintf("/accounts/%s/customer-managed-keys/%s",
+		mwsAcctID, customerManagedKeyID), nil, &k)
+	return
+}
+
+// Delete deletes the customer managed key object given a network id
+func (a CustomerManagedKeysAPI) Delete(mwsAcctID, customerManagedKeyID string) error {
+	return a.client.Delete(fmt.Sprintf("/accounts/%s/customer-managed-keys/%s",
+		mwsAcctID, customerManagedKeyID), nil)
+}
+
+// List lists all the available customer managed key objects in the mws account
+func (a CustomerManagedKeysAPI) List(mwsAcctID string) (kl []CustomerManagedKey, err error) {
+	err = a.client.Get(fmt.Sprintf("/accounts/%s/customer-managed-keys", mwsAcctID), nil, &kl)
+	return
+}
+
+var customerManagedKeySchema = resourceMWSCustomerManagedKeysSchema()
+
+func ResourceCustomerManagedKey() *schema.Resource {
+	return &schema.Resource{
+		SchemaVersion: 2,
+		Create:        resourceCustomerManagedKeyCreate,
+		Read:          resourceCustomerManagedKeyRead,
+		Delete:        resourceCustomerManagedKeyDelete,
+		Schema:        customerManagedKeySchema,
+	}
+}
+
+func resourceMWSCustomerManagedKeysSchema() map[string]*schema.Schema {
```

Comment:
why do we need a function if it's called only 3 lines of code and called once for package-local variable?

Reply:
its a function because it is used in two other locations in the file. Alternatively I can make it:

```go
var customerManagedKeySchema = internal.StructToSchema(CustomerManagedKey{}, func(s map[string]*schema.Schema) map[string]*schema.Schema {
		s["aws_key_info"].ForceNew = true
		s["account_id"].ForceNew = true
		return s
	})
```

Reply:
let me know what you would prefer.

Reply:
if it's called two times, then there's no need to change :)